src/Features/Core/Portable/Formatting/FormattingCodeFixProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = PredefinedCodeFixProviderNames.FixFormatting)]
+    [Shared]
+    internal class FormattingCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.FormattingDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(c => FixOneAsync(context.Document, diagnostic, c)),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected async Task<Document> FixOneAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+
+            // The span to format is the full line(s) containing the diagnostic
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var diagnosticLinePositionSpan = text.Lines.GetLinePositionSpan(diagnosticSpan);
+            var spanToFormat = TextSpan.FromBounds(
+                text.Lines[diagnosticLinePositionSpan.Start.Line].Start,
+                text.Lines[diagnosticLinePositionSpan.End.Line].End);
```

Comment:
i'm a little confused... why do we nee dthis code, if we have the CodeStyle version that does all ths...

Reply:
Extracted to a shared helper class `FormattingCodeFixHelper`